packages/web-components/src/components/masthead/left-nav-menu-highlighted.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { customElement } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import DDSLeftNavMenu from './left-nav-menu';
+import styles from './masthead.scss';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Masthead left nav submenu item.
+ *
+ * @element dds-left-nav-menu-item
+ */
+@customElement(`${ddsPrefix}-left-nav-menu-highlighted`)
+class DDSLeftNavMenuHighlighted extends DDSLeftNavMenu {
+  static styles = styles; // `styles` here is a `CSSResult` generated by custom WebPack loader
+}
```

Comment:
My knee-jerk reaction here was adding prop for highlighted state rather than a new component, but looking at `<dds-masthead-composite>` code, it seems something different from highlighted state for the nav? Do you have any details?

Reply:
Yeah I tried adding a `highlighted` attribute to the `dds-left-nav-menu` component but styling wise for the mobile menu I need to target the last highlighted nav menu to set the darker border and 
`:host(#{dds-prefix}-left-nav-menu)[highlighted]:last-of-type` didn't work as the `:last-of-type` or `:last-child` selector does not take attributes into account

Reply:
Thank you for elaborating @annawen1! Just in case - Did you try `:host(#{dds-prefix}-left-nav-menu[highlighted]):last-of-type`? I think there are other CSS rules whose selector matches host element with certain attributes, and I suggest to look at those.

Reply:
@asudoh yes that works! I didn't think to wrap the attribute within the `:host()` 😅. This is great, I'll remove the unnecessary components. Thanks!

Reply:
@asudoh I might have spoken too quickly...I tested the above by just setting the attribute `highlighted` without any conditionals `<dds-left-nav-menu highlighted .... >` and it worked. But as soon as I add a conditional whether to render the attribute `<dds-left-nav-menu-item ?highlighted=${ifNonNull(item.highlighted)} ...>` then I don't see the styles at applied at all. I tried setting a class depending on whether it is highlighted or not but that doesn't seem to work either

Reply:
Thank you for the update @annawen1 - One quick comment is we don't need `ifNonNull()` for boolean attributes, because `lit-html` automatically unsets the attribute if the value is falsy.

For the issue you described, would you inspect the DOM and see what CSS ruleset isn't applied while it should? If you cannot figure it out, probably you can push the code you tried (but it doesn't work)?

Reply:
@asudoh sure, I can try pushing the code.

When I inspect, I am expecting to see this CSS applied to the last `left-nav-menu[highlighted]`element
```
  :host(#{$dds-prefix}-left-nav-menu[highlighted]):last-of-type {
    border-bottom: 1px solid $ui-04;
  }
```
<img width="1562" alt="Screen Shot 2021-01-29 at 8 53 48 AM" src="https://user-images.githubusercontent.com/54281166/106283425-ea4edb00-620f-11eb-95a5-4659d97a3965.png">

Reply:
@annawen1 Thank you for elaborating! Seems that the CSS rule works if I add `highlighted` attribute to the `<dds-left-nav-menu>` for "Explore more" at the top-level. One thing I imagined the menu you brought up is not really a `:last-of-type` - As long as what the DOM inspector shows.

Reply:
@asudoh Right it works when the attribute is added to the last `nav-menu` of the section, if I try to add it to "Learn & Support" it doesn't work. What I'm trying to do is target the last `dds-left-nav-menu` with the `highlighted` attribute.

Reply:
I see - BTW what does "highlighted" mean? I'm asking this because adding `highlighted` attribute doesn't actually seem to make the menu highlighted. Any context...?

Reply:
The `highlighted` is to note that the menu item is part of the "highlighted section" - the left nav section in desktop and with the new designs, the highlighted items have a distinct border to separate between the highlighted menu items). 

<img width="1636" alt="Screen Shot 2021-01-27 at 2 04 04 PM" src="https://user-images.githubusercontent.com/54281166/106045139-98427400-60ae-11eb-99e7-0004248b4ed9.png">

<img width="348" alt="Screen Shot 2021-01-27 at 2 04 25 PM" src="https://user-images.githubusercontent.com/54281166/106045163-9e385500-60ae-11eb-9e35-ce033a9a3d01.png">

https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/5016/iframe.html?id=components-masthead--with-custom-data

Reply:
@annawen1 I see thanks for clarifying! Given you brought up "highlighted section" words, my suggestion is creating a component for such "highlighted section".

Reply:
@asudoh yeah that makes sense - I'll try that out, thanks!

Reply:
So I tried created a component (`dds-left-nav-highlighted`) to wrap the `dds-left-nav-menu` items into the "highlighted section" but ran into the same issue with the `:last-of-type` or `:last-child` selector. I still need to target the last `dds-left-nav-menu` item to remove its own border.

```
 :host(#{$dds-prefix}-left-nav-highlighted) {
    display: block;
    border-bottom: 1px solid $ui-04;

    ::slotted(#{$dds-prefix}-left-nav-menu:last-of-type) .#{$prefix}--side-nav__submenu::after {
      content: none;
    }
  }
```

Doesn't seem like I'm able to target the last `dds-left-nav-menu` within the hightlighted section here either. 

You can see the border of the last menu item is still showing here:
<img width="444" alt="Screen Shot 2021-01-29 at 1 28 27 PM" src="https://user-images.githubusercontent.com/54281166/106315594-76292d00-6239-11eb-8dfa-ea2c4af2f235.png">

Reply:
I ended up just setting the `last-highlighted` attribute to the last highlighted menu item of the section and styling with that instead. That avoids creating new components and doesn't require us to use the `:last-of-type` selector!

Reply:
@annawen1 Thank you for explaining what you stumbled at! My suggestion is adding `::after` pseudo element to `<dds-left-nav-highlighted>` with something like below, so the border there is placed on top of (in z-axis) the bottom border of the last `<dds-left-nav-menu>`:

```css
:host(#{$dds-prefix}-left-nav-highlighted) {
  position: relative;

  /* (Some other rules) */

  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    border-bottom: 1px solid $ui-04;
    width: 100%;
  }
}
```

Reply:
@asudoh so I also attempted to create a wrapper container in React and another issue I found is that because the submenus structurally are within each other, if I add a border to the wrapper container and then click into "Hybrid Cloud" submenu, the border remains for the next menu section which is something we don't want. It seems to be more trouble to set when to hide the border in this case

Reply:
Thank you for taking deeper look @annawen1! I see that `position:relative` at the wrapper container changes what `min-height:100%` at `<ul class="bx--side-nav__menu">` in `<dds-left-nav>`'s shadow DOM refers to, that breaks its feature of "covering up parent menu items". So I think we'd go with your solution of adding attribute to the last item in the "highlighted" group.